metadata-ingestion/src/datahub/ingestion/source/grafana/lineage.py

```diff
@@ -29,6 +30,99 @@
 
 logger = logging.getLogger(__name__)
 
+# Precompiled regex patterns for Grafana template variable cleaning
+# These patterns remove Grafana-specific template syntax to make SQL parseable
+
+# Time/filter macros with parentheses: $__timeFilter(column), $__timeGroup(...)
+# Replace with TRUE as they form complete boolean expressions
+_GRAFANA_TIME_MACRO_WITH_ARGS_PATTERN = re.compile(r"\$__time[A-Z]\w*\([^)]*\)")
+_GRAFANA_FILTER_MACRO_WITH_ARGS_PATTERN = re.compile(r"\$__[a-z]+Filter\([^)]*\)")
+
+# Time/filter macros WITHOUT parentheses (standalone): $__timeFilter, $__interval
+# These are used as predicates and need to be replaced with valid SQL expressions
+_GRAFANA_TIME_MACRO_STANDALONE_PATTERN = re.compile(r"\$__time[A-Z]\w*(?!\()")
+_GRAFANA_FILTER_MACRO_STANDALONE_PATTERN = re.compile(r"\$__[a-z]+Filter(?!\()")
+
+# Generic macros (with or without args): $__interval, $__range, etc.
+_GRAFANA_GENERIC_MACRO_PATTERN = re.compile(r"\$__\w+(?:\([^)]*\))?")
+
+# Bracket and braced variables
+_GRAFANA_BRACKET_VAR_PATTERN = re.compile(r"\[\[[^\]]+\]\]")
+_GRAFANA_BRACED_VAR_PATTERN = re.compile(r"\$\{[^}]+\}")
+
+# Simple variables NOT inside quotes: $var
+# Use negative lookbehind/lookahead to skip variables already in quotes
+_GRAFANA_SIMPLE_VAR_PATTERN = re.compile(r"(?<!')(\$[a-zA-Z_][a-zA-Z0-9_]*)(?!')")
```

Comment:
even if grafana is not supported dialect in sqlglot, have we checked whether sqlglot could make some of this cleaning?